### PR TITLE
Add future warning for focal loss shape and clarify use in documentat…

### LINKED ISSUE #1242: Clarify/make consistent expected prediction output shape for different loss functions
For classification tasks, cross entropy will have a prediction shape with number of columns that's equal to the number of classes. So for binary tasks we have 2 columns in the output, where col2 = 1-col1.

Focal loss instead will produce one column that is the positive score, which carries the same information while being more space-efficient. 

We need to inform users and possibly enforce a consistent behavior for num_clases

### PR DESCRIPTION
**Pull request by thvasilo**
*Issue #, if available:*

Fixes #1242 

*Description of changes:*

* Add FutureWarning at object creation time for FocalLoss and warning in the documentation that its output is about to change in 0.5.0

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 1]
```diff
@@ -452,11 +452,21 @@ Classification and Regression Task
     - Yaml: ``save_prediction_path: /data/infer-output/predictions/``
     - Argument: ``--save-prediction-path /data/infer-output/predictions/``
     - Default value: If not provided, it will be the same as save_embed_path.
-- **class_loss_func**: Node/Edge classification loss function. Builtin loss functions include ``cross_entropy`` and ``focal``. ``focal`` means to use the focal loss function defined in the `Focal Loss for Dense Object Detection <https://arxiv.org/abs/1708.02002>`_, which is designed for class imbalance. If set ``focal``, you may want to adjust the values of the **gamma** and **alpha** configurations according to your data.
+- **class_loss_func**: Node/Edge classification loss function. Builtin loss functions include ``cross_entropy`` and ``focal``.
+  Setting this to ``focal`` will user the focal loss function defined in `Focal Loss for Dense Object Detection <https://arxiv.org/abs/1708.02002>`_,
```
Comment (classicsong) [inline docs/source/cli/model-training-inference/configuration-run.rst:456]:
```suggestion
  Setting this to ``focal`` will use the focal loss function defined in `Focal Loss for Dense Object Detection <https://arxiv.org/abs/1708.02002>`_,
```

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 2]
```diff
@@ -452,11 +452,21 @@ Classification and Regression Task
...
 
     - Yaml: ``class_loss_func: cross_entropy``
     - Argument: ``--class-loss-func focal``
     - Default value: ``cross_entropy``
+
+    .. note::
+
+        Focal loss can only be used for binary classification problems.
+        Currently, focal loss produces predictions with shape ``(N, 1)``, where ``N`` is the number of target nodes/edges.
+        In ``v0.5.0`` this will change to produce predictions with shape ``(N, 2)`` to match the cross-entropy loss.
```
Comment (classicsong) [inline docs/source/cli/model-training-inference/configuration-run.rst:468]:
```suggestion
        In ``v0.5.0`` this may be changed to produce predictions with shape ``(N, 2)`` to match the cross-entropy loss.
```

**Review by classicsong — APPROVED** [review]
